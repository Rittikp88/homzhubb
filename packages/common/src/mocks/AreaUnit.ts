export const AreaUnit = [
  {
    label: 'Sq.ft',
    value: 'Sq.ft',
  },
  {
    label: 'Sq.yard',
    value: 'Sq.yard',
  },
  {
    label: 'Sq.m',
    value: 'Sq.m',
  },
  {
    label: 'cents',
    value: 'cents',
  },
  {
    label: 'biswa',
    value: 'biswa',
  },
  {
    label: 'acres',
    value: 'acres',
  },
  {
    label: 'hectares',
    value: 'hectares',
  },
  {
    label: 'kattha',
    value: 'kattha',
  },
];
