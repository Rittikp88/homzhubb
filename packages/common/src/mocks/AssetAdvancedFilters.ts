export const AdvancedFilters = {
  searchRadius: [
    {
      value: 1,
      label: 'This area only',
    },
    {
      value: 2,
      label: 'Within 0.25 miles',
    },
    {
      value: 3,
      label: 'Within 0.5 miles',
    },
    {
      value: 4,
      label: 'Within 1 miles',
    },
    {
      value: 5,
      label: 'Within 3 miles',
    },
    {
      value: 6,
      label: 'Within 10 miles',
    },
    {
      value: 7,
      label: 'Within 20 miles',
    },
    {
      value: 8,
      label: 'Within 30 miles',
    },
    {
      value: 9,
      label: 'Within 40 miles',
    },
  ],
  dateAdded: [
    {
      value: 1,
      label: 'Anytime',
    },
    {
      value: 2,
      label: 'Last 3 days',
    },
    {
      value: 3,
      label: 'Last 14 days',
    },
    {
      value: 4,
      label: 'Last 24 hours',
    },
    {
      value: 5,
      label: 'Last 7 days',
    },
    {
      value: 6,
      label: 'Last 28 days',
    },
  ],
  propertyAge: [
    {
      value: 1,
      label: 'Less than 1 year',
    },
    {
      value: 2,
      label: 'Less than 5 years',
    },
    {
      value: 3,
      label: 'Less than 10 years',
    },
    {
      value: 4,
      label: 'Less than 15 years',
    },
    {
      value: 5,
      label: 'Less than 20 years',
    },
    {
      value: 6,
      label: 'More than 20 years',
    },
  ],
  moveInDate: '',
  facing: [
    {
      value: 1,
      title: 'East',
    },
    {
      value: 2,
      title: 'West',
    },
    {
      value: 3,
      title: 'North',
    },
    {
      value: 4,
      title: 'South',
    },
    {
      value: 5,
      title: 'North-East',
    },
    {
      value: 6,
      title: 'South-east',
    },
    {
      value: 7,
      title: 'North-West',
    },
    {
      value: 8,
      title: 'South-West',
    },
    {
      value: 9,
      title: 'Any value',
    },
  ],
  furnishing: [
    {
      value: 1,
      label: 'Television',
    },
    {
      value: 2,
      label: 'Iron',
    },
    {
      value: 3,
      label: 'Washing Machine',
    },
    {
      value: 4,
      label: 'Geysers',
    },
    {
      value: 5,
      label: 'Microwave',
    },
    {
      value: 6,
      label: 'Modular Kitchen',
    },
    {
      value: 7,
      label: 'Beds',
    },
    {
      value: 8,
      label: 'Air-Purifier',
    },
    {
      value: 9,
      label: 'Mattress',
    },
    {
      value: 10,
      label: 'Air Conditioner',
    },
  ],
  propertyAmenities: [
    {
      value: 1,
      label: 'Swimming Pool',
    },
    {
      value: 2,
      label: 'Near Metro station',
    },
    {
      value: 3,
      label: 'Gym',
    },
    {
      value: 4,
      label: 'Tennis court',
    },
    {
      value: 5,
      label: 'Daily needs store',
    },
    {
      value: 6,
      label: 'Table tennis',
    },
    {
      value: 7,
      label: 'Near ATM',
    },
    {
      value: 8,
      label: 'Club-house',
    },
    {
      value: 9,
      label: 'Near petrol pump',
    },
    {
      value: 10,
      label: 'Helipad',
    },
  ],
  isAgentListed: false,
  isVerified: false,
};
