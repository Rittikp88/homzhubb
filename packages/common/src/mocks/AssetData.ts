import { theme } from '@homzhub/common/src/styles/theme';
import { images } from '@homzhub/common/src/assets/images';

export interface IAssetData {
  id: number;
  property_name: string;
  address: string;
  type: string;
  color: string;
  images: any;
  contacts?: any;
  isPropertyCompleted: boolean;
}

export const TenanciesAssetData = [
  {
    id: 5,
    project_name: 'House 2',
    unit_number: '103',
    block_number: 'B',
    latitude: 25.6207722,
    longitude: 85.1280892,
    carpet_area: 2500.0,
    carpet_area_unit: 'SQ_FT',
    floor_number: 2,
    total_floors: 3,
    asset_type: {
      id: 5,
      name: 'Studio Apartment',
    },
    spaces: [
      {
        id: 1,
        name: 'Bedroom',
        count: 2,
      },
      {
        id: 2,
        name: 'Bathroom',
        count: 1,
      },
    ],
    asset_group: {
      id: 1,
      name: 'Residential',
    },
    digital_id: '8fad8389-6893-411a-bd4a-4b1575523d4a',
    construction_year: null,
    is_gated: null,
    description: '',
    furnishing: 'SEMI',
    attachments: [
      {
        file_name: 'prof.jpg',
        is_cover_image: true,
        link: 'https://homzhub-bucket.s3.amazonaws.com/asset_images/8e8c48fc-c089-11ea-8247-34e12d38d70eprof.jpg',
        media_type: 'IMAGE',
        media_attributes: {},
      },
    ],
    verification_documents: [],
    verifications: {
      verification_status: 2,
      description:
        '**What is this verification?**\nSed ut perspiciatis unde omnis iste natus error sit voluptatem\naccusantium doloremque laudantium.\n\nNemo enim ipsam voluptatem quia voluptas sit aspernatur\naut odit aut fugit. Neque porro quisquam est,\nqui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit,\nsed quia non numquam eius modi tempora incidunt ut labore et dolore\nmagnam aliquam quaerat voluptatem.',
    },
    progress_percentage: 74.0,
    notifications: {
      count: 5,
    },
    serviceTickets: {
      count: 0,
    },
    assetStatusInfo: {
      tag: {},
      leaseTransaction: {},
      leaseTenantInfo: {},
    },
  },
];

export const PortfolioAssetData = [
  {
    id: 1,
    property_name: '2BHK - Godrej Prime',
    address: 'Sindhi Society, Chembur, Mumbai- 400071',
    type: 'OWNER',
    isPropertyCompleted: true,
    color: '#CE9B6C',
    images: [
      {
        file_name: 'House-1b',
        is_cover_image: false,
        link: images.property,
      },
    ],
    contacts: {
      id: 1,
      full_name: 'Anuvrat Somnath',
      email: 'anuvrat.somnath@nineleaps.com',
      country_code: '+91',
      phone_number: '7903470293',
    },
  },
  {
    id: 2,
    property_name: '2BHK - Godrej Prime',
    address: 'Sindhi Society, Chembur, Mumbai- 400071',
    type: 'RENTED',
    isPropertyCompleted: true,
    color: theme.colors.gradientE,
    images: [
      {
        file_name: 'House-1b',
        is_cover_image: false,
        link: images.property,
      },
    ],
    contacts: {
      id: 1,
      full_name: 'Anuvrat Somnath',
      email: 'anuvrat.somnath@nineleaps.com',
      country_code: '+91',
      phone_number: '7903470293',
    },
  },
  {
    id: 3,
    property_name: '2BHK - Godrej Prime',
    address: 'Sindhi Society, Chembur, Mumbai- 400071',
    type: 'VACANT',
    isPropertyCompleted: true,
    color: theme.colors.highPriority,
    images: [
      {
        file_name: 'House-1b',
        is_cover_image: false,
        link: images.property,
      },
    ],
    contacts: {
      id: 1,
      full_name: 'Anuvrat Somnath',
      email: 'anuvrat.somnath@nineleaps.com',
      country_code: '+91',
      phone_number: '7903470293',
    },
  },
  {
    id: 4,
    property_name: '2BHK - Godrej Prime',
    address: 'Sindhi Society, Chembur, Mumbai- 400071',
    type: 'VACANT',
    isPropertyCompleted: false,
    color: theme.colors.highPriority,
    images: [
      {
        file_name: 'House-1b',
        is_cover_image: false,
        link: images.property,
      },
    ],
  },
  {
    id: 5,
    property_name: '2BHK - Godrej Prime',
    address: 'Sindhi Society, Chembur, Mumbai- 400071',
    type: 'FOR SALE',
    isPropertyCompleted: true,
    color: theme.colors.mediumPriority,
    images: [
      {
        file_name: 'House-1b',
        is_cover_image: false,
        link: images.property,
      },
    ],
    contacts: {
      id: 1,
      full_name: 'Anuvrat Somnath',
      email: 'anuvrat.somnath@nineleaps.com',
      country_code: '+91',
      phone_number: '7903470293',
    },
  },
  {
    id: 6,
    property_name: '2BHK - Godrej Prime',
    address: 'Sindhi Society, Chembur, Mumbai- 400071',
    type: 'MAINTAIN',
    isPropertyCompleted: true,
    color: theme.colors.informational,
    images: [
      {
        file_name: 'House-1b',
        is_cover_image: false,
        link: images.property,
      },
    ],
    contacts: {
      id: 1,
      full_name: 'Anuvrat Somnath',
      email: 'anuvrat.somnath@nineleaps.com',
      country_code: '+91',
      phone_number: '7903470293',
    },
  },
];

export const AssetFilter = [
  {
    label: 'All',
    value: 'all',
  },
  {
    label: 'Rented',
    value: 'rented',
  },
  {
    label: 'Vacant',
    value: 'vacant',
  },
  {
    label: 'For Sale',
    value: 'sale',
  },
  {
    label: 'Maintain',
    value: 'maintain',
  },
];

export const DocumentsData = [
  {
    id: 1,
    name: 'Quarterly Inspection Report',
    uploaded_on: '02/07/2020',
    uploaded_by: 'HomzHub',
  },
  {
    id: 2,
    name: 'Rent',
    uploaded_on: '02/07/2020',
    uploaded_by: 'HomzHub',
  },
  {
    id: 3,
    name: 'Electricity bill',
    uploaded_on: '02/07/2020',
    uploaded_by: 'HomzHub',
  },
  {
    id: 4,
    name: 'Water charges',
    uploaded_on: '02/07/2020',
    uploaded_by: 'HomzHub',
  },
  {
    id: 5,
    name: 'Maintenance Charges',
    uploaded_on: '02/07/2020',
    uploaded_by: 'HomzHub',
  },
];
