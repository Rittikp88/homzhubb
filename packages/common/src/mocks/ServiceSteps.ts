enum ServiceStepTypes {
  LEASE_DETAILS = 'LEASE_DETAILS',
  PROPERTY_IMAGES = 'PROPERTY_IMAGES',
  PROPERTY_VERIFICATIONS = 'PROPERTY_VERIFICATIONS',
  PAYMENT_TOKEN_AMOUNT = 'PAYMENT_TOKEN_AMOUNT',
}
export const ServiceSteps = [
  {
    id: 1,
    name: ServiceStepTypes.LEASE_DETAILS,
    title: 'Enter the Lease details',
  },
  {
    id: 2,
    name: ServiceStepTypes.PROPERTY_IMAGES,
    title: 'Add Property Images',
  },
  {
    id: 3,
    name: ServiceStepTypes.PROPERTY_VERIFICATIONS,
    title: 'Complete property verification',
  },
  {
    id: 4,
    name: ServiceStepTypes.PAYMENT_TOKEN_AMOUNT,
    title: 'Payment of token amount',
  },
];
