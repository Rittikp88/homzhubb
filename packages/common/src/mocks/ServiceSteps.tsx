export const ServiceSteps = [
  {
    name: 'Free Listing',
    steps: [
      { id: 1, name: 'Enter the Lease details' },
      { id: 2, name: 'Add Property Images' },
    ],
  },
  {
    name: 'Assisted Listing',
    steps: [
      { id: 1, name: 'Enter the Lease details' },
      { id: 2, name: 'Add Property Images' },
      { id: 3, name: 'Complete property verification' },
      { id: 4, name: 'Payment of token amount' },
    ],
  },
  {
    name: 'Boosted Listing',
    steps: [
      { id: 1, name: 'Enter the Lease details' },
      { id: 2, name: 'Add Property Images' },
      { id: 3, name: 'Complete property verification' },
      { id: 4, name: 'Payment of token amount' },
    ],
  },
];
