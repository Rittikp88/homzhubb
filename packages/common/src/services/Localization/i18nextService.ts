import i18next, { i18n } from 'i18next';
import { initReactI18next } from 'react-i18next';
import { I18nManager } from 'react-native';
import languageDetector from '@homzhub/common/src/services/Localization/languageDetector';
import { LocaleConstants, supportedLanguages } from '@homzhub/common/src/services/Localization/constants';
import { Logger } from '@homzhub/common/src/utils/Logger';

class I18nextService {
  private readonly _instance: i18n;

  constructor() {
    this._instance = i18next.createInstance();
  }

  public init = async (): Promise<void> => {
    this._instance.use(initReactI18next).use(languageDetector);
    await this._instance.init(
      {
        whitelist: LocaleConstants.whitelist,
        resources: LocaleConstants.resources,
        fallbackLng: LocaleConstants.fallback,
        ns: LocaleConstants.namespaces,
        defaultNS: LocaleConstants.defaultNamespace,
        interpolation: {
          escapeValue: false,
        },
      },
      this.initCallback
    );
  };

  get instance(): i18n {
    return this._instance;
  }

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  public t = (key: string, options?: any): string => {
    if (!this._instance) {
      this.init().then();
    }
    return this._instance.t(key, options);
  };

  public isRTL = (): boolean => {
    return I18nManager.isRTL;
  };

  public select<T>(map: { [position in 'rtl' | 'ltr']: T }): T {
    const key = this.isRTL() ? 'rtl' : 'ltr';
    return map[key];
  }

  public changeLanguage = async (lngCode: supportedLanguages): Promise<void> => {
    await this._instance.changeLanguage(lngCode);
  };

  private initCallback = (error: Error): void => {
    if (error) {
      Logger.error(error);
    } else {
      Logger.info('Loaded i18next');
    }
  };
}

const i18nService = new I18nextService();
export { i18nService as I18nService };
