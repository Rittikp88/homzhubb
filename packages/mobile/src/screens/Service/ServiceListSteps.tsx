import React from 'react';
import { StyleSheet, View } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { bindActionCreators, Dispatch } from 'redux';
import { connect } from 'react-redux';
import { theme } from '@homzhub/common/src/styles/theme';
import { icons } from '@homzhub/common/src/assets/icon';
import { IState } from '@homzhub/common/src/modules/interfaces';
import { PropertyActions } from '@homzhub/common/src/modules/property/actions';
import { PropertySelector } from '@homzhub/common/src/modules/property/selectors';
import { IServiceListStepsDetail } from '@homzhub/common/src/domain/models/Service';
import { Label, Text, Divider } from '@homzhub/common/src/components';
import Header from '@homzhub/mobile/src/components/molecules/Header';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { AppStackParamList } from '@homzhub/mobile/src/navigation/AppNavigator';

interface IStateProps {
  serviceSteps: IServiceListStepsDetail[];
}

interface IDispatchProps {
  getServiceStepsDetails: (id: number) => void;
}

type libraryProps = WithTranslation & NavigationScreenProps<AppStackParamList, ScreensKeys.ServiceListSteps>;
type Props = libraryProps & IStateProps & IDispatchProps;

class ServiceListSteps extends React.PureComponent<Props, {}> {
  public componentDidMount(): void {
    const {
      getServiceStepsDetails,
      route: { params },
    } = this.props;
    getServiceStepsDetails(params.id);
  }

  public render(): React.ReactNode {
    const {
      t,
      route: { params },
    } = this.props;
    return (
      <View style={styles.container}>
        <Header
          backgroundColor={theme.colors.primaryColor}
          icon={icons.leftArrow}
          iconColor="white"
          iconStyle={styles.iconStyle}
          onIconPress={this.navigateBack}
          isHeadingVisible
          title={t('service:listProperty')}
          titleType="regular"
          titleFontType="semiBold"
          titleStyle={styles.headerTitle}
        />
        <View style={styles.listing}>
          <Text type="regular" textType="semiBold" style={styles.listingTitle}>
            {params.name}
          </Text>
          <Label type="large" textType="semiBold" style={styles.label} onPress={this.navigateBack}>
            {t('common:change')}
          </Label>
        </View>
        <View style={styles.initiateService}>
          <Text type="small" textType="semiBold" style={styles.subHeader}>
            {t('service:subHeader')}
          </Text>
          {this.renderSteps()}
        </View>
      </View>
    );
  }

  public renderSteps(): React.ReactNode {
    const { serviceSteps } = this.props;
    return serviceSteps.map((stepItem: IServiceListStepsDetail, index: number) => {
      const isLast = serviceSteps.length - 1 === index;
      return (
        <>
          <View key={index} style={styles.stepView}>
            <Text type="small" textType="regular" style={styles.stepItem}>
              {index + 1}
            </Text>
            <Text type="regular" textType="regular" style={styles.stepName}>
              {stepItem.title}
            </Text>
          </View>
          {!isLast && <Divider containerStyles={styles.divider} key={`divider-${index}`} />}
        </>
      );
    });
  }

  public navigateBack = (): void => {
    const { navigation } = this.props;
    navigation.goBack();
  };
}

const mapStateToProps = (state: IState): IStateProps => {
  const { getServiceSteps } = PropertySelector;
  return {
    serviceSteps: getServiceSteps(state),
  };
};

const mapDispatchToProps = (dispatch: Dispatch): IDispatchProps => {
  const { getServiceStepsDetails } = PropertyActions;
  return bindActionCreators(
    {
      getServiceStepsDetails,
    },
    dispatch
  );
};

export default connect<IStateProps, IDispatchProps, WithTranslation, IState>(
  mapStateToProps,
  mapDispatchToProps
)(withTranslation()(ServiceListSteps));

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: theme.colors.white,
  },
  iconStyle: {
    flex: 1,
  },
  headerTitle: {
    color: theme.colors.white,
  },
  listing: {
    flexDirection: 'row',
    margin: theme.layout.screenPadding,
  },
  listingTitle: {
    flex: 1,
  },
  label: {
    color: theme.colors.primaryColor,
  },
  initiateService: {
    margin: theme.layout.screenPadding,
  },
  subHeader: {
    color: theme.colors.darkTint4,
  },
  stepView: {
    flexDirection: 'row',
    justifyContent: 'center',
    marginTop: 25,
  },
  stepItem: {
    width: 35,
    height: 35,
    borderColor: theme.colors.darkTint4,
    borderRadius: 35,
    borderWidth: 2,
    paddingTop: 5,
    textAlign: 'center',
  },
  divider: {
    borderColor: theme.colors.disabled,
    borderWidth: 1,
    width: 35,
    marginTop: 25,
    transform: [{ rotate: '90deg' }],
  },
  stepName: {
    flex: 1,
    marginLeft: 20,
  },
});
