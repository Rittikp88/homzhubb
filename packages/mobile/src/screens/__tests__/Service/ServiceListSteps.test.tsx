import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { I18nextProvider } from 'react-i18next';
import { I18nService } from '@homzhub/common/src/services/Localization/i18nextService';
import { ServiceSteps } from '@homzhub/common/src/mocks/ServiceSteps';
import ServiceListSteps from '@homzhub/mobile/src/screens/Service/ServiceListSteps';

jest.mock('@homzhub/common/src/services/storage/StorageService', () => 'StorageService');
jest.mock('@react-native-community/google-signin');
jest.mock('@homzhub/common/src/components/', () => 'Label');
jest.mock('@homzhub/common/src/components/', () => 'Text');
jest.mock('@homzhub/mobile/src/components/molecules/Header', () => 'Header');

const mockStore = configureStore([]);

describe('Service List Steps Screen', () => {
  let store: any;
  let component: any;
  let props: any;
  beforeEach(async () => {
    store = mockStore({
      service: {
        servicesSteps: ServiceSteps[0].steps,
      },
    });
    props = {
      navigation: {
        navigate: jest.fn(),
      },
    };
    await I18nService.init();
    component = shallow(
      <Provider store={store}>
        <I18nextProvider i18n={I18nService.instance}>
          <ServiceListSteps {...props} />
        </I18nextProvider>
      </Provider>
    );
  });
  it('should render snapshot', () => {
    expect(toJson(component.dive().dive().dive())).toMatchSnapshot();
  });
});
