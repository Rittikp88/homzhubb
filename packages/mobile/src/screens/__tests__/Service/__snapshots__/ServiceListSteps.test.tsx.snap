// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service List Steps Screen should render snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Header
      backgroundColor="#0084F8"
      icon="left-arrow"
      iconColor="white"
      iconStyle={
        Object {
          "flex": 1,
        }
      }
      isHeadingVisible={true}
      onIconPress={[Function]}
      title="property:listProperty"
      titleFontType="semiBold"
      titleStyle={
        Object {
          "color": "#FFFFFF",
        }
      }
      titleType="regular"
    />
    <ScrollViewMock
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "margin": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "flex": 1,
            }
          }
          textType="semiBold"
          type="regular"
        >
          Some Listing
        </Text>
        <Label
          onPress={[Function]}
          style={
            Object {
              "color": "#0084F8",
            }
          }
          testID="lblNavigate"
          textType="semiBold"
          type="large"
        >
          common:change
        </Label>
      </View>
      <View
        style={
          Object {
            "margin": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#4C5F72",
            }
          }
          textType="semiBold"
          type="small"
        >
          property:subHeader
        </Text>
        <View
          key="stepContainer0"
        >
          <View
            key="0"
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 25,
              }
            }
          >
            <View
              key="0"
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#4C5F72",
                  "borderRadius": 17.5,
                  "borderWidth": 2,
                  "height": 35,
                  "justifyContent": "center",
                  "width": 35,
                }
              }
            >
              <Text
                key="stepNo0"
                textType="regular"
                type="small"
              >
                1
              </Text>
            </View>
            <Text
              adjustsFontSizeToFit={true}
              key="stepTitle0"
              minimumFontScale={0.5}
              numberOfLines={1}
              style={
                Object {
                  "alignSelf": "center",
                  "flex": 1,
                  "marginLeft": 20,
                }
              }
              textType="regular"
              type="small"
            >
              Enter the Lease details
            </Text>
          </View>
          <Divider
            containerStyles={
              Object {
                "borderColor": "#C9D6DF",
                "borderWidth": 1,
                "marginTop": 25,
                "transform": Array [
                  Object {
                    "rotate": "90deg",
                  },
                ],
                "width": 35,
              }
            }
            key="divider-0"
          />
        </View>
        <View
          key="stepContainer1"
        >
          <View
            key="1"
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 25,
              }
            }
          >
            <View
              key="1"
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#4C5F72",
                  "borderRadius": 17.5,
                  "borderWidth": 2,
                  "height": 35,
                  "justifyContent": "center",
                  "width": 35,
                }
              }
            >
              <Text
                key="stepNo1"
                textType="regular"
                type="small"
              >
                2
              </Text>
            </View>
            <Text
              adjustsFontSizeToFit={true}
              key="stepTitle1"
              minimumFontScale={0.5}
              numberOfLines={1}
              style={
                Object {
                  "alignSelf": "center",
                  "flex": 1,
                  "marginLeft": 20,
                }
              }
              textType="regular"
              type="small"
            >
              Add Property Images
            </Text>
          </View>
        </View>
      </View>
    </ScrollViewMock>
  </View>
  <WithShadowView
    outerViewStyle={
      Object {
        "marginBottom": 20,
        "paddingBottom": 0,
        "paddingTop": 10,
      }
    }
  >
    <Button
      containerStyle={
        Object {
          "flex": 0,
          "margin": 16,
        }
      }
      onPress={[Function]}
      testID="btnContinue"
      title="common:getStarted"
      type="primary"
    />
  </WithShadowView>
</View>
`;

exports[`Service List Steps Screen should render snapshot 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Header
      backgroundColor="#0084F8"
      icon="left-arrow"
      iconColor="white"
      iconStyle={
        Object {
          "flex": 1,
        }
      }
      isHeadingVisible={true}
      onIconPress={[Function]}
      title="property:listProperty"
      titleFontType="semiBold"
      titleStyle={
        Object {
          "color": "#FFFFFF",
        }
      }
      titleType="regular"
    />
    <ScrollViewMock
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "margin": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "flex": 1,
            }
          }
          textType="semiBold"
          type="regular"
        >
          Some Listing
        </Text>
        <Label
          onPress={[Function]}
          style={
            Object {
              "color": "#0084F8",
            }
          }
          testID="lblNavigate"
          textType="semiBold"
          type="large"
        >
          common:change
        </Label>
      </View>
      <View
        style={
          Object {
            "margin": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#4C5F72",
            }
          }
          textType="semiBold"
          type="small"
        >
          property:subHeader
        </Text>
        <View
          key="stepContainer0"
        >
          <View
            key="0"
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 25,
              }
            }
          >
            <View
              key="0"
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#4C5F72",
                  "borderRadius": 17.5,
                  "borderWidth": 2,
                  "height": 35,
                  "justifyContent": "center",
                  "width": 35,
                }
              }
            >
              <Text
                key="stepNo0"
                textType="regular"
                type="small"
              >
                1
              </Text>
            </View>
            <Text
              adjustsFontSizeToFit={true}
              key="stepTitle0"
              minimumFontScale={0.5}
              numberOfLines={1}
              style={
                Object {
                  "alignSelf": "center",
                  "flex": 1,
                  "marginLeft": 20,
                }
              }
              textType="regular"
              type="small"
            >
              Enter the Lease details
            </Text>
          </View>
          <Divider
            containerStyles={
              Object {
                "borderColor": "#C9D6DF",
                "borderWidth": 1,
                "marginTop": 25,
                "transform": Array [
                  Object {
                    "rotate": "90deg",
                  },
                ],
                "width": 35,
              }
            }
            key="divider-0"
          />
        </View>
        <View
          key="stepContainer1"
        >
          <View
            key="1"
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 25,
              }
            }
          >
            <View
              key="1"
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#4C5F72",
                  "borderRadius": 17.5,
                  "borderWidth": 2,
                  "height": 35,
                  "justifyContent": "center",
                  "width": 35,
                }
              }
            >
              <Text
                key="stepNo1"
                textType="regular"
                type="small"
              >
                2
              </Text>
            </View>
            <Text
              adjustsFontSizeToFit={true}
              key="stepTitle1"
              minimumFontScale={0.5}
              numberOfLines={1}
              style={
                Object {
                  "alignSelf": "center",
                  "flex": 1,
                  "marginLeft": 20,
                }
              }
              textType="regular"
              type="small"
            >
              Add Property Images
            </Text>
          </View>
          <Divider
            containerStyles={
              Object {
                "borderColor": "#C9D6DF",
                "borderWidth": 1,
                "marginTop": 25,
                "transform": Array [
                  Object {
                    "rotate": "90deg",
                  },
                ],
                "width": 35,
              }
            }
            key="divider-1"
          />
        </View>
        <View
          key="stepContainer2"
        >
          <View
            key="2"
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 25,
              }
            }
          >
            <View
              key="2"
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#4C5F72",
                  "borderRadius": 17.5,
                  "borderWidth": 2,
                  "height": 35,
                  "justifyContent": "center",
                  "width": 35,
                }
              }
            >
              <Text
                key="stepNo2"
                textType="regular"
                type="small"
              >
                3
              </Text>
            </View>
            <Text
              adjustsFontSizeToFit={true}
              key="stepTitle2"
              minimumFontScale={0.5}
              numberOfLines={1}
              style={
                Object {
                  "alignSelf": "center",
                  "flex": 1,
                  "marginLeft": 20,
                }
              }
              textType="regular"
              type="small"
            >
              Complete property verification
            </Text>
          </View>
          <Divider
            containerStyles={
              Object {
                "borderColor": "#C9D6DF",
                "borderWidth": 1,
                "marginTop": 25,
                "transform": Array [
                  Object {
                    "rotate": "90deg",
                  },
                ],
                "width": 35,
              }
            }
            key="divider-2"
          />
        </View>
        <View
          key="stepContainer3"
        >
          <View
            key="3"
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 25,
              }
            }
          >
            <View
              key="3"
              style={
                Object {
                  "alignItems": "center",
                  "borderColor": "#4C5F72",
                  "borderRadius": 17.5,
                  "borderWidth": 2,
                  "height": 35,
                  "justifyContent": "center",
                  "width": 35,
                }
              }
            >
              <Text
                key="stepNo3"
                textType="regular"
                type="small"
              >
                4
              </Text>
            </View>
            <Text
              adjustsFontSizeToFit={true}
              key="stepTitle3"
              minimumFontScale={0.5}
              numberOfLines={1}
              style={
                Object {
                  "alignSelf": "center",
                  "flex": 1,
                  "marginLeft": 20,
                }
              }
              textType="regular"
              type="small"
            >
              Payment of token amount
            </Text>
          </View>
        </View>
      </View>
    </ScrollViewMock>
  </View>
  <WithShadowView
    outerViewStyle={
      Object {
        "marginBottom": 20,
        "paddingBottom": 0,
        "paddingTop": 10,
      }
    }
  >
    <Button
      containerStyle={
        Object {
          "flex": 0,
          "margin": 16,
        }
      }
      onPress={[Function]}
      testID="btnContinue"
      title="common:getStarted"
      type="primary"
    />
  </WithShadowView>
</View>
`;
